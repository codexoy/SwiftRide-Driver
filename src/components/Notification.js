import React, { Fragment } from "react"
import { StyleSheet, View, Text } from "react-native"
import UserWithPic from "./UserWithPic"
import Row from "./Row"
import { strings } from "../i18n"
import Separator from "./Separator"

const extras = {
	new_lesson: strings("teacher.notifications.new_lesson")
}

export default class Notification extends React.Component {
	render() {
		return (
			<Fragment>
				<Row
					style={this.props.style}
					leftSide={
						<View style={styles.dateAndHour}>
							<Text style={styles.date}>{this.props.date}</Text>
							<Text style={styles.hour}>{this.props.hours}</Text>
						</View>
					}
				>
					<UserWithPic
						name={this.props.name}
						extra={<Text>{extras[this.props.type]}</Text>}
						width={54}
						height={54}
					/>
				</Row>
				<View style={styles.buttons}>{this.props.children}</View>
				<Separator />
			</Fragment>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	dateAndHour: { flex: 1, marginTop: 5 },
	date: {
		fontSize: 22,
		fontFamily: "Assistant-Light"
	},
	hour: {
		fontWeight: "bold",
		fontSize: 14,
		color: "rgb(12, 116, 244)"
	},
	buttons: {
		flex: 1,
		marginTop: 16,
		flexDirection: "row",
		justifyContent: "space-between"
	}
})
