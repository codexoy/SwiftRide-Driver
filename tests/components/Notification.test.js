import { Text } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import Notification from "../../src/components/Notification"

describe("Notification", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<Notification
					style={{ height: 30 }}
					name="test"
					type="new_lesson"
					date="test"
					hours="test-test"
				>
					<Text>Hello</Text>
				</Notification>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
