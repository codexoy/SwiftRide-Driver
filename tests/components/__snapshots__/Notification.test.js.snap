// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification view renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Image
          source={
            Object {
              "uri": "https://images.unsplash.com/photo-1535643302794-19c3804b874b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2134&q=80",
            }
          }
          style={
            Object {
              "borderRadius": 27,
              "height": 54,
              "width": 54,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          test
        </Text>
        <Text>
          רוצה לקבוע שיעור
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.5,
          "marginRight": "auto",
          "marginTop": -8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Assistant-Light",
              "fontSize": 22,
            }
          }
        >
          test
        </Text>
        <Text
          style={
            Object {
              "color": "rgb(12, 116, 244)",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          test-test
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 16,
      }
    }
  >
    <Text>
      Hello
    </Text>
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": "rgb(212, 212, 212)",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 20,
        "width": "100%",
      }
    }
  />,
]
`;
